bigquery_etl/docs/derived_datasets/generate_derived_dataset_docs.py

```diff
@@ -58,6 +60,23 @@ def generate_derived_dataset_docs(out_dir, project_dir):
                 if VIEW_FILE in files:
                     source_urls["View Definition"] = f"{SOURCE_URL}/{root}/{VIEW_FILE}"
 
+                    view_file = os.path.join(root, VIEW_FILE)
+                    content = ""
+
+                    with open(view_file) as f:
+                        view_content = f.read()
+                        # get the view content by removing CREATE OR REPLACE VIEW
```

Comment:
Why do we need to strip the CREATE OR REPLACE VIEW part? Do we not get table references when dry running a DML statement?

Reply:
I believe because the current dry run service only has READ permission so if a query has "CREATE OR REPLACE VIEW" it will return a permission error. We have been bypassing this error here:
https://github.com/mozilla/bigquery-etl/blob/5148a7f775a968e1d46205b67cca160daa762ae7/bigquery_etl/dryrun.py#L239-L251

> If there's a way that we can improve the dry run service to be able to return the info we need from CREATE VIEW statements and avoid the need to manipulate the content, I would prefer to do that.

It definitely sounds like a better approach. I'm not familiar enough with the cloud run service to know if this is possible, any advice @scholtzan @acmiyaguchi?

Reply:
Thank you for the specific link. I had forgotten about the read-only restriction, which indeed complicates matters. Maybe we can still figure out a workaround.

Reply:
The error message makes sense. I actually don't understand why we need the DML statement in the first place (I don't need them in terraform or the console to create views in tables). 

I mentioned an alternative in a previous comment, which is to use `select * from {view} where {partition filter}`. This assumes the view has already been deployed, but does not require modifying the SQL itself.

Reply:
> I actually don't understand why we need the DML statement in the first place (I don't need them in terraform or the console to create views in tables).

I agree that it would be better not to have DML in our `.sql` files, but it would be some significant effort to change the design at this point.